paimon-spark/paimon-spark-common/src/main/scala/org/apache/paimon/spark/SparkTable.scala

```diff
@@ -60,6 +60,8 @@ case class SparkTable(table: Table)
           properties.put(CoreOptions.PRIMARY_KEY.key, String.join(",", table.primaryKeys))
         }
         properties.put(TableCatalog.PROP_PROVIDER, SparkSource.NAME)
+        val location = properties.remove(CoreOptions.PATH.key())
```

Comment:
This interface (`default Map<String, String> properties()`) is only used in show create or desc table, add the `TableCatalog.PROP_LOCATION` is ok, but wondering whether we need to remove `path`.

Besides, another point worth noting is that paimon currently does not support `create table x location('')` actually?  The new result of `show create table` may cause misunderstanding,  or we need support it first.

Reply:
Hi @zhongyujiang , we can discuss in this thread, @Zouxxyy raised two very good points:
1. We could keep `path` option.
2. We should also support execution for `show create table`.

Reply:
>  We could keep path option.

Sure

> We should also support execution for show create table

I've tested this and found that the location info passed by Spark DDL will be ignored right now

IIUC, the table location is closely related to the warehouse address, and of course, different catalogs have different implementations:

- AbstractCatalog calculates the table location based on the warehouse, database, and table name when loading a table, and both FileSystemCatalog and JdbcCatalog follow this behavior
- HiveCatalog retrieves the table location information from the HMS (Hive Metastore)

Therefore, if we want to support this, we need to differentiate the catalog being used. This is feasible with HiveCatalog, but not with other catalogs. What do you think? @Zouxxyy @JingsongLi

Reply:
For other catalogs, we can check the location, if not equal, throw exception.

Reply:
can see #https://github.com/apache/paimon/issues/2619

Reply:
I have updated this to allow customizing the table location when using the Hive catalog and creating an external table. Please take a look when you have time. Thanks!